graphql_compiler/tests/schema_transformation_tests/test_rename_schema.py

```diff
@@ -21,6 +22,38 @@
 from .input_schema_strings import InputSchemaStrings as ISS
 
 
+def validate_error_message_component(
+    error_message_component: str, expected_prefix: str, expected_data_structure: Any
+) -> bool:
+    """Confirm that error_message_component is as expected.
+
+    A common pattern when checking string representations of exceptions here is to have the message
+    consist of a string literal prefix, which describes the error reasoning, followed by the string
+    representation of a data structure that describes which parts of the schema caused the error.
+    This function validates the error message based on the expected prefix and expected data
+    structure. This is useful when the data structure's __str__() method is not deterministic
+    (e.g. set).
+
+    Args:
+        error_message_component: error message part to validate
+        expected_prefix: explanation of why the error was raised
+        expected_data_structure: description of which parts of the schema caused the error
+
+    Returns:
+        True iff the error message correctly consists of the prefix followed by the expected data
+        structure.
+    """
+    if not error_message_component.startswith(expected_prefix):
+        return False
+    try:
+        if literal_eval(error_message_component[len(expected_prefix) :]) != expected_data_structure:
+            return False
+    except SyntaxError:
+        # In case it's syntactically invalid
+        return False
+    return True
+
+
 def check_rename_conflict_error_message(
```

Comment:
Having revised the data structures to check for no-op renames, I removed the big method checking that string for the `NoOpRenamingError`. Unfortunately, since name conflicts have a more complicated structure (i.e. it's not just "renamings contains these keys which do nothing"), and since it's easier to generate many name conflicts accidentally (as opposed to manually creating a renamings dict with many unused entries), I'm not sure about just sorting everything in the error message. Thoughts?

Reply:
I'm not sure what the current structure is that is more complicated. I agree that there could be many name conflicts — the canonical solution to that is to check the list length and if too large, only output a few elements (~7ish or so) from the head of the sorted list, and a few (maybe ~3) from the tail. This keeps the error message small. Then, you can add an instance attribute to the error type that contains the full list of errors, allowing callers to see and recover the full list even though it's not in the error message.

Reply:
Ah, gotcha-- to clarify, this is the order of events that happened:

* PR #884 gets merged (for `SchemaRenameNameConflictError`, which has a complicated error message because the data structures are involved are complicated).

* This PR comes out (for `NoOpRenamingError`, which also had a complicated error message that's constructed in a similar way as `SchemaRenameNameConflictError`).

* Since the two error types have similarly constructed error messages, I decide to abstract out some of it into this `validate_error_message_component` function.

* I find a way to simplify the data structure for `NoOpRenamingError` so only `SchemaRenameNameConflictError` tests use `validate_error_message_component` function now.

In the process of writing this out I rubber-duck-debugged myself into realizing that this means any code involving `validate_error_message_component` no longer has to do with this PR's purpose since `NoOpRenamingError` tests don't rely on complicated string checks anymore, so I'll revert those changes and split into a separate PR as appropriate.